api-client/lib/api/extensions/get_into_teaching_api_client/api_client.rb

```diff
@@ -33,27 +42,30 @@ def call_api(http_method, path, opts = {})
           req.headers["Content-Type"] = "application/json"
         end
 
-        unless response.success?
-          raise ::GetIntoTeachingApiClient::ApiError.new(
-            code: response.status,
-            response_headers: response.headers,
-            response_body: response.body,
-          )
-        end
-
         if opts[:return_type]
           data = deserialize(response, opts[:return_type])
         end
 
         [data, response.status, response.headers]
+
+      rescue Faraday::Error => error
+        if error.response.present?
+          raise ::GetIntoTeachingApiClient::ApiError.new(
+            code: error.response_status,
+            response_headers: error.response_headers,
+            response_body: error.response_body,
+          )
+        else
+          raise error
```

Comment:
I think we're missing test coverage for this line; it would be worth adding a quick test (not circuit breaker related but in with the others) that ensures faraday errors that don't have a response get re-raised